.template/lib/di/interceptor/app_interceptor.dart

```diff
@@ -15,4 +17,25 @@ class AppInterceptor extends Interceptor {
     }
     return super.onRequest(options, handler);
   }
+
+  @override
+  void onError(DioError err, ErrorInterceptorHandler handler) {
+    final statusCode = err.response?.statusCode;
+    if ((statusCode == HttpStatus.forbidden ||
+        statusCode == HttpStatus.unauthorized) &&
+        _requireAuthenticate) {
+      _doRefreshToken(err, handler);
+    } else {
+      handler.next(err);
+    }
+  }
+
+  Future<void> _doRefreshToken(
+      DioError err,
+      ErrorInterceptorHandler handler,
+      ) async {
+    //TODO handle refresh token here
+
```

Comment:
should we prepare some logic here for faster & consistent implementation in the real project? Some action could be in TODO are:
- Request new token from an API request
- Update new token header
How do you think?

```suggestion
import 'package:varun_matchmaking_flutter/usecases/base/base_use_case.dart';

  try {
    // TODO Request new token
    
    if (result is Success) {
      // TODO Update new token header
      // err.requestOptions.headers[_headerAuthorization] = newToken;

      // Create request with new access token
      final options = Options(
          method: err.requestOptions.method,
          headers: err.requestOptions.headers);
      final newRequest = await _dio.request(
          "${err.requestOptions.baseUrl}${err.requestOptions.path}",
          options: options,
          data: err.requestOptions.data,
          queryParameters: err.requestOptions.queryParameters);
      handler.resolve(newRequest);
    } else {
      handler.next(err);
    }
  } catch (exception) {
    if (exception is DioError) {
      handler.next(exception);
    } else {
      handler.next(err);
    }
  }
```

Reply:
@luongvo Good idea, however, it will be costly as we have to define the usecase here.
I suggest adding the comment code so that we can have it for future reference

Reply:
@doannimble as it's the template, I don't expect to add any use case integration here. Please don't misunderstand here as I mentioned there are 2 TODO (same as my comments in suggestion code). So in the real project implementation, the developer will need to implement their code at:
- `// TODO Request new token`
- `// TODO Update new token header`

 How do you think? 💭

Reply:
@luongvo Make sense, I've updated 🙏